contracts/sol6/Dao/mock/KyberFeeHandlerWrapper.sol

```diff
@@ -0,0 +1,228 @@
+        require(_kyberDao != IKyberDao(0), "kyberDao 0");
+        kyberDao = _kyberDao;
+    }
+
+    function addFeeHandler(IERC20 _token, ExtendedIKyberFeeHandler _kyberFeeHandler) external onlyDaoOperator {
+        addTokenToSupportedTokensArray(_token);
+        addFeeHandlerToKyberFeeHandlerArray(kyberFeeHandlersPerToken[_token], _kyberFeeHandler);
+        emit FeeHandlerAdded(_token, _kyberFeeHandler);
+    }
+
+    /// @dev claim from multiple feeHandlers
+    /// @param staker staker address
+    /// @param epoch epoch for which the staker is claiming the reward
+    /// @param startTokenIndex index of supportedTokens to start iterating from (inclusive)
+    /// @param endTokenIndex index of supportedTokens to start iterating from (exclusive)
+    /// @param startKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (inclusive)
+    /// @param endKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (exclusive)
+    /// @return amounts staker reward wei / twei amount claimed from each feeHandler
+    function claimStakerReward(
+        address staker,
+        uint256 epoch,
+        uint256 startTokenIndex,
+        uint256 endTokenIndex,
+        uint256 startKyberFeeHandlerIndex,
+        uint256 endKyberFeeHandlerIndex
+    ) external returns(uint256[] memory amounts) {
+        if (
+            startTokenIndex > endTokenIndex ||
+            startKyberFeeHandlerIndex > endKyberFeeHandlerIndex ||
+            supportedTokens.length == 0
+        ) {
+            // no need to do anything
+            return amounts;
+        }
+
+        uint256 endTokenId = (endTokenIndex >= supportedTokens.length) ?
+            supportedTokens.length : endTokenIndex;
+
+        for (uint256 i = startTokenIndex; i < endTokenId; i++) {
+            KyberFeeHandlerData[] memory kyberFeeHandlerArray = kyberFeeHandlersPerToken[supportedTokens[i]];
+            uint256 endKyberFeeHandlerId = (endKyberFeeHandlerIndex >= kyberFeeHandlerArray.length) ?
+                kyberFeeHandlerArray.length - 1: endKyberFeeHandlerIndex;
+            amounts = new uint256[](endKyberFeeHandlerId - startKyberFeeHandlerIndex + 1);
+
+            // iteration starts from endIndex, differs from claiming reserve rebates and platform wallets
+            for (uint256 j = endKyberFeeHandlerId; j >= startKyberFeeHandlerIndex; j--) {
+                KyberFeeHandlerData memory kyberFeeHandlerData = kyberFeeHandlerArray[j];
+                if (kyberFeeHandlerData.startEpoch < epoch) {
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
+                    break;
+                } else if (kyberFeeHandlerData.startEpoch == epoch) {
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
+                } else if (j == 0) {
+                    break;
+                } else {
+                    continue;
+                }
```

Comment:
Redundant `continue` here?

Reply:
If `kyberFeeHandlerArray[0].startEpoch == epoch`, this for loop will run forever?

Reply:
so if (j ==0) should probably be without else

Reply:
Yes, the `continue` is redundant, but I wanted to be explicit (make it clear). Can remove.... lmk wad u think